services:
  modelio:
    image: fazenda/modelio
    restart: unless-stopped
    environment:
      - TZ=America/Sao_Paulo
      - USER_ID=0
      - GROUP_ID=0
    volumes:
      - ${PWD}/workspace:/root/modelio/workspace
    labels:
      - traefik.enable=true
      - traefik.http.routers.modelio.tls=true
      - traefik.http.routers.modelio.rule=Host(`modelio.docker.localhost`)
    networks:
      - proxy

  reverse-proxy:
    image: traefik:v3.3.6
    security_opt:
      - no-new-privileges:true
    restart: on-failure
    ports:
      - 80:80
      - 443:443
      - 8080:8080
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ${HOME}/.certs:/etc/certs:ro
      - ${HOME}/.traefik/traefik.yml:/etc/traefik/traefik.yml:ro
      - ${HOME}/.traefik/config.yml:/etc/traefik/config.yml:ro
    labels:
      - traefik.enable=true
      - traefik.http.routers.traefik=true
    networks:
      - proxy

networks:
  proxy:
    external: true
